src/hooks/useTxTracking.ts

```diff
@@ -1,23 +1,43 @@
 import { trackEvent, WALLET_EVENTS } from '@/services/analytics'
+import { getTxDetails } from '@/services/tx/txDetails'
 import { TxEvent, txSubscribe } from '@/services/tx/txEvents'
 import { useEffect } from 'react'
+import useChainId from './useChainId'
 
 const events = {
-  [TxEvent.SIGNED]: WALLET_EVENTS.OFF_CHAIN_SIGNATURE,
-  [TxEvent.PROCESSING]: WALLET_EVENTS.ON_CHAIN_INTERACTION,
-  [TxEvent.RELAYING]: WALLET_EVENTS.RELAYED_EXECUTION,
+  [TxEvent.SIGNED]: WALLET_EVENTS.OFFCHAIN_SIGNATURE,
+  [TxEvent.PROCESSING]: WALLET_EVENTS.ONCHAIN_INTERACTION,
+  [TxEvent.PROCESSING_MODULE]: WALLET_EVENTS.ONCHAIN_INTERACTION,
+  [TxEvent.RELAYING]: WALLET_EVENTS.ONCHAIN_INTERACTION,
 }
 
 export const useTxTracking = (): void => {
+  const chainId = useChainId()
+
   useEffect(() => {
     const unsubFns = Object.entries(events).map(([txEvent, analyticsEvent]) =>
-      txSubscribe(txEvent as TxEvent, () => {
-        trackEvent(analyticsEvent)
+      txSubscribe(txEvent as TxEvent, async (detail) => {
+        const txId = 'txId' in detail ? detail.txId : undefined
+        const txHash = 'txHash' in detail ? detail.txHash : undefined
+        const id = txId || txHash
+
+        let origin = ''
+        if (id) {
+          try {
+            const txDetails = await getTxDetails(chainId, id)
+            origin = txDetails.safeAppInfo?.url || ''
```

Comment:
Should we track the wallet type as a fallback here i.e. `wallet.label`?

Reply:
The (peer) label may be a better option as we have to [use the WC Safe App URL as the `appUrl`](https://github.com/safe-global/safe-wallet-web/pull/2641/files#diff-63ad7ff9d4750580543f5094dab28fde1143f63658e694c67eaa680a6cc2f998R91) to get the WC icon in the transaction list.